acisfp_check/acis_obs.py

```diff
@@ -40,11 +39,61 @@ def who_in_fp(simpos=80655):
     return is_in_the_fp
 
 
-# ---------------------------------------------------------------------
-#
-# find_obsid_intervals
-#
-# ---------------------------------------------------------------------
+def fetch_ocat_data(obsid_list):
+    """
+    Take a list of obsids and return the following data from
+    the obscat for each: grating status, CCD count, if S3 is on,
+    and the number of expected counts
+
+    Parameters
+    ----------
+    obsid_list : list of ints
+        The obsids to get the obscat data from.
+
+    Returns
+    -------
+    A dict of NumPy arrays of the above properties.
+    """
+    import requests
+    from astropy.io import ascii
+    urlbase = "https://cda.harvard.edu/srservices/ocatDetails.do?format=text"
+    obsid_list = ",".join([str(obsid) for obsid in obsid_list])
+    params = {"obsid": obsid_list}
+    # First fetch the information from the obsid itself
+    resp = requests.get(urlbase, params=params)
+    tab = ascii.read(resp.text, header_start=0, data_start=2)
+    tab["TSTART"] = CxoTime(tab["START_DATE"].data).secs
+    tab.sort("TSTART")
+    # We figure out the CCD count from the table by finding out
+    # which ccds were on or optional, and then subtracting off
+    # the dropped chip count
+    ccd_count = np.zeros(tab["S3"].size, dtype='int')
```

Comment:
Maybe misreading things but it looks to me like ccd_count is an array with length of your N_obsid, but a few lines down it appears you don't index into the array in adding / subtracting to ccd_count

Reply:
You are correct about the shape of the ccd_count array. All of the operations done to this array in the following lines are vectorized themselves, so we don't need to do indexing.